modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js

```diff
@@ -104,22 +157,85 @@ const Modal = ({bodyHTML, buttons, id, onClose, size, title, url}) => {
 		}
 	};
 
-	const BodyHTML = () => {
+	const Body = ({html}) => {
 		const bodyRef = useRef();
 
 		useEffect(() => {
 			const fragment = document
 				.createRange()
-				.createContextualFragment(bodyHTML);
+				.createContextualFragment(html);
 
 			bodyRef.current.innerHTML = '';
 
 			bodyRef.current.appendChild(fragment);
-		}, []);
+		}, [html]);
 
 		return <div ref={bodyRef}></div>;
 	};
 
+	class Iframe extends React.Component {
+		constructor(props) {
+			super(props);
+
+			this.iframeRef = React.createRef();
+
+			const iframeURL = new URL(props.url);
+
+			const namespace = iframeURL.searchParams.get('p_p_id');
+
+			let bodyCssClass = 'dialog-iframe-popup';
+
+			if (iframeBodyCssClass) {
+				bodyCssClass = `${bodyCssClass} ${iframeBodyCssClass}`;
+			}
+
+			iframeURL.searchParams.set(
+				`_${namespace}_bodyCssClass`,
+				bodyCssClass
+			);
+
+			this.state = {src: iframeURL.toString()};
+		}
+
+		onLoadHandler = () => {
+			console.log('iframe loaded');
+
+			const iframe = this.iframeRef.current;
+
+			dom.delegate(
+				iframe.contentWindow.document,
+				'click',
+				'.btn-cancel',
+				() => processClose()
+			);
+
+			setLoading(false);
```

Comment:
This is suspicious (using a hook that you don't "own" from a class you defined inside the closure of a function component). The `class` has to be defined at the top level of the module because otherwise you'll re-define it on every render, which I think is why your `shouldComponentUpdate` isn't working.

Reply:
Yep, that did it! Awesome! I'll followup with PR soon.